generic/tests/trans_hugepage.py

```diff
@@ -64,33 +64,33 @@ def get_mem_status(params, role):
     logging.info("Smoke test start")
     error_context.context("smoke test")
 
-    nr_ah_before = int(get_mem_status('AnonHugePages', 'host'))
+    nr_ah_before = get_mem_status('AnonHugePages', 'host')
     if nr_ah_before <= 0:
         e_msg = 'smoke: Host is not using THP'
         logging.error(e_msg)
         failures.append(e_msg)
 
     # Protect system from oom killer
-    if int(get_mem_status('MemFree', 'guest')) / 1024 < mem:
-        mem = int(get_mem_status('MemFree', 'guest')) / 1024
+    if get_mem_status('MemFree', 'guest') / 1024 < mem:
+        mem = int(get_mem_status('MemFree', 'guest') / 1024)
 
     session.cmd("mkdir -p %s" % mem_path)
 
     session.cmd("mount -t tmpfs -o size=%sM none %s" % (str(mem), mem_path))
 
-    count = mem / 4
+    count = int(mem / 4)
     session.cmd("dd if=/dev/zero of=%s/1 bs=4000000 count=%s" %
                 (mem_path, count), timeout=dd_timeout)
 
-    nr_ah_after = int(get_mem_status('AnonHugePages', 'host'))
+    nr_ah_after = get_mem_status('AnonHugePages', 'host')
 
     if nr_ah_after <= nr_ah_before:
         e_msg = ('smoke: Host did not use new THP during dd')
         logging.error(e_msg)
         failures.append(e_msg)
 
     if debugfs_flag == 1:
-        if int(open('%s/kvm/largepages' % debugfs_path, 'r').read()) <= 0:
+        if int(open('%s/kvm/lpages' % debugfs_path, 'r').read()) <= 0:
```

Comment:
Tested with RHEL8.6, this path already change from the `largepages` to `lpages`,  I hardcode this path as a temporary fix. 

@huangyum Could you help confirm this change ? May be we need define it for different product, thanks!

Reply:
@mcasquer Would you please help confirm this issue and review this patch? Thanks.

Reply:
> @mcasquer Would you please help confirm this issue and review this patch? Thanks.

Sure!

Reply:
Code reviewed and test case passed, see my [comment](https://github.com/autotest/tp-qemu/pull/3128#issuecomment-1022132828)

Reply:
@yanan-fu is there a plan to change lpages to pages_2m [as its default HP size]? or just read all pages_<size> ->add up and check if that's <=0?